go/adbc/driver/bigquery/driver.go

```diff
@@ -96,18 +135,35 @@ func init() {
 
 type driverImpl struct {
 	driverbase.DriverImplBase
+	clientFactory      bigqueryClientFactory
+	tokenSourceFactory impersonatedTokenSourceFactory
 }
 
 // NewDriver creates a new BigQuery driver using the given Arrow allocator.
 func NewDriver(alloc memory.Allocator) adbc.Driver {
+	return NewDriverWithClientFactory(alloc, &defaultBigqueryClientFactory{})
+}
+
+// NewDriverWithClientFactory creates a new BigQuery driver using the given Arrow allocator and client factory.
+func NewDriverWithClientFactory(alloc memory.Allocator, clientFactory bigqueryClientFactory) adbc.Driver {
+	if clientFactory == nil {
+		clientFactory = &defaultBigqueryClientFactory{}
+	}
+	return NewDriverWithClientAndTokenSourceFactory(alloc, clientFactory, &defaultImpersonatedTokenSourceFactory{})
+}
+
+// NewDriverWithClientAndTokenSourceFactory creates a new BigQuery driver using the given Arrow allocator, client factory, and impersonated token source factory.
+func NewDriverWithClientAndTokenSourceFactory(alloc memory.Allocator, clientFactory bigqueryClientFactory, tokenSourceFactory impersonatedTokenSourceFactory) adbc.Driver {
```

Comment:
Ditto here, there are private types in the signature, the function should be private